File: develnext-designer/src/main/resources/JPHP-INF/sdk/php/gui/dock/UXDockTitleBar.php
<?php
namespace php\gui\dock;

use php\gui\layout\UXHBox;
use php\gui\layout\UXVBox;
use php\gui\UXButton;
use php\gui\UXLabel;
use php\gui\UXNode;

/**
 * @package php\gui\dock
 */
class UXDockTitleBar extends UXHBox
{
    /**
     * @var UXLabel
     */
    public $label;

    /**
     * @readonly
     * @var UXButton
     */
    public $closeButton;

    /**
     * @readonly
     * @var UXButton
     */
    public $stateButton;

    /**
     * @readonly
     * @var UXButton
     */
    public $minimizeButton;

    /**
     * UXDockTitleBar constructor.
     * @param UXDockNode $node
     */
    public function __construct(UXDockNode $node)
    {
    }

    /**
     * @return bool
     */
    public function isDragging()
    {
    }
}

File: develnext-designer/src/main/java/org/develnext/jphp/gui/designer/classes/dock/UXDockTitleBar.java
package org.develnext.jphp.gui.designer.classes.dock;

import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.HBox;
import org.develnext.jphp.ext.javafx.classes.UXNode;
import org.develnext.jphp.ext.javafx.classes.layout.UXHBox;
import org.develnext.jphp.gui.designer.GuiDesignerExtension;
import org.dockfx.DockTitleBar;
import php.runtime.annotation.Reflection;
import php.runtime.annotation.Reflection.Property;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Reflection.Namespace(GuiDesignerExtension.NS_DOCK)
public class UXDockTitleBar extends UXHBox<DockTitleBar> {
    interface WrappedInterface {
        @Property Label label();

        @Property Button closeButton();
        @Property Button minimizeButton();
        @Property Button stateButton();

        boolean isDragging();
    }

    public UXDockTitleBar(Environment env, DockTitleBar wrappedObject) {
        super(env, wrappedObject);
    }

    public UXDockTitleBar(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct(UXDockNode node) {
        __wrappedObject = new DockTitleBar(node.getWrappedObject());
    }
}
